### PR TITLE
Add linting in GitHub Actions

### PR DESCRIPTION
**Pull request by nikosavola**
### Description

This PR runs `eslint` in GitHub Actions to enforce code style

### Resolved / Related Issues

_None_

### Type of Change

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation
- [ ] Localization 

### Validation Checklist
None of these really apply

- [ ] Built and ran the app
- [ ] Tested my changes _(to make sure your changes don't break something, make sure to test the code)_
- [ ] Formatted code using Prettier/ESLint
- [ ] Added comments/documentation in relevant places _([JSDocs](https://jsdoc.app/) for function/class body and `//` comments for other parts)_

### CODE REVIEW

**Comment by nikosavola** [comment]
The Action works but the linting itself fails due to the code not following the given [`.eslintrc.json`](https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/blob/master/.eslintrc.json), see https://github.com/nikosavola/Google-Assistant-Unofficial-Desktop-Client/runs/5396248190?check_suite_focus=true

**Comment by Melvin-Abraham** [comment]
Thank you for the contribution, @nikosavola. Based on the logs provided, most of the error seems to pop up due to the external library, `.../lib/pv_workers`. Try adding this directory to `ignorePatterns` field in `.eslintrc.json` file.

**Comment by nikosavola** [comment]
> Thank you for the contribution, @nikosavola. Based on the logs provided, most of the error seems to pop up due to the external library, `.../lib/pv_workers`. Try adding this directory to `ignorePatterns` field in `.eslintrc.json` file.

With this change, there are only a few warnings. Currently, only errors fail the workflow, not warnings. 

Should this be kept this way or changed?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -36,3 +36,28 @@ jobs:
 
       - name: Build
         run: npm run dist:ci
+
+  lint:
+
```
Comment (Melvin-Abraham) [inline .github/workflows/main.yml:41]: nit: This extra blank line could be removed

**Comment by Melvin-Abraham** [comment]
> Currently, only errors fail the workflow, not warnings.
>
> Should this be kept this way or changed?

@nikosavola This is the expected behavior. It can stay that way...

**Comment by Melvin-Abraham** [comment]
There seems to be an issue in the pipeline due to certain breaking changes introduced by GitHub for security as stated in <https://github.blog/2021-09-01-improving-git-protocol-security-github/>. I'll try to make the required changes to fix them...

**Review by Melvin-Abraham — APPROVED** [review]
LGTM

**Pull request merged**